openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -205,6 +209,25 @@ def celebrations(self):
             'first_section': CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object),
         }
 
+    @property
+    def user_has_passing_grade(self):
+        course = get_course_by_id(self.course_key)
+        user_grade = CourseGradeFactory().read(self.request.user, course).percent
+
+        return user_grade >= course.lowest_passing_grade
+
+    @property
+    def course_completion_is_active(self):
+        return course_completion_is_active(self.course_key)
+
+    @property
+    def certificate_data(self):
+        course = get_course_with_access(self.request.user, 'load', self.course_key, check_if_enrolled=True)
```

Comment:
Do you need to use the `get_course_with_access` here? It looks like a lot of the other properties just use   `get_course_by_id`? 
There also seems to already been an enrollment_object defined as part of `self`. Maybe you could use that to grab the enrollment mode. Might help simplify this function.